docs/content/latest/api/ysql/the-sql-language/statements/ddl_alter_table.md

```diff
@@ -76,7 +88,112 @@ Renaming a table is a non blocking metadata change operation.
 
 Drop the named column from the table. 
 
-- `RESTRICT` — Remove only the specified
+- `RESTRICT` — Remove only the specified column
+- `CASCADE` — Remove the specified column and any dependent objects.
+
+##### Example
+
+Set up a and populate a parents-children pair of tables thus:
+
+```plpgssql
+drop table if exists children cascade;
+drop table if exists parents  cascade;
+
+-- The column "b" models a (natural) business unique key.
+create table parents(
+  k int primary key,
+  b int not null,
+  v text not null,
+  constraint parents_b_unq unique(b));
+
+create table children(
+  parents_b  int  not null,
+  k         int  not null,
+  v         text not null,
+
+  constraint children_pk primary key(parents_b, k),
+
+  constraint children_fk foreign key(parents_b)
+    references parents(b)
+    match full
+    on delete cascade
+    on update restrict);
+
+insert into parents(k, b, v) values (1, 10, 'dog'), (2, 20, 'cat'), (3, 30, 'frog');
+
+insert into children(parents_b, k, v) values
+  (10, 1, 'dog-child-a'),
+  (10, 2, 'dog-child-b'),
+  (10, 3, 'dog-child-c'),
+  (20, 1, 'cat-child-a'),
+  (20, 2, 'cat-child-b'),
+  (20, 3, 'cat-child-c'),
+  (30, 1, 'frog-child-a'),
+  (30, 2, 'frog-child-b'),
+  (30, 3, 'frog-child-c');
+
+select p.v as "p.v", c.v as "c.v"
+from parents p inner join children c on c.parents_b = p.b
+order by p.b, c.k;
+```
+
+This is the result:
+
+```output
+ p.v  |     c.v      
+------+--------------
+ dog  | dog-child-a
+ dog  | dog-child-b
+ dog  | dog-child-c
+ cat  | cat-child-a
+ cat  | cat-child-b
+ cat  | cat-child-c
+ frog | frog-child-a
+ frog | frog-child-b
+ frog | frog-child-c
+```
+
+The `\d children` metacommand shows that it has a foreign key that's a dependent object on the column `b` in the `parents` table thus:
+
+```output
+Indexes:
+    "children_pk" PRIMARY KEY, lsm (parents_b HASH, k ASC)
+Foreign-key constraints:
+    "children_fk" FOREIGN KEY (parents_b) REFERENCES parents(b) MATCH FULL ON UPDATE RESTRICT ON DELETE CASCADE
+```
+
+This is a contrived example. It is unusual practice (and normally bad practice) to make a foreign key constraint target anything but the column list upon which the parent table's primary key constraint is defined. But there are sometimes defensible reasons to do this.
```

Comment:
Worth mentioning what those sometimes-defensible reasons are here, or no?

Reply:
I decided not too. It's a whole, fairly big, topic in database design. Best to be brief—but to call out that YB isn't recommending something that appears a bit odd.